packages/blob/src/index.node.test.ts

```diff
@@ -148,9 +150,9 @@ describe('blob client', () => {
         })
         .reply(502, { error: { code: 'service_unavailable' } });
 
-      await expect(head(`${BLOB_STORE_BASE_URL}/foo-id.txt`)).rejects.toThrow(
-        new BlobServiceNotAvailable(),
-      );
+      await expect(
+        head(`${BLOB_STORE_BASE_URL}/foo-id.txt`, { retries: 0 }),
```

Comment:
Here, if you want to test that, at some point we will receive `BlobServiceNotAvailable`, then there should be a way to have the Mock consistently fail, unless I am missing something?